File: Python/791. Custom Sort String.py
class Solution:
    def customSortString(self, S, T):
        """
        :type S: str
        :type T: str
        :rtype: str
        """
        maple = dict()
        vis, cnt = [25]*26, 0
        tt = [0]*26
        for c in S:
            vis[ord(c) - ord('a')] = cnt
            cnt += 1
        for c in T:
            tmp = vis[ord(c) - ord('a')]
            if tt[tmp] == 0:
                tt[tmp] = c
            else:
                tt[vis[ord(c) - ord('a')]] += c
        ans = ''
        for i in range(26):
            tmp = tt[i]
            if tt[i] != 0:
                ans += ''.join(tt[i])
        return ans
            

File: C++/791. Custom Sort String.cpp
class Solution {
public:
    string customSortString(string S, string T) {
        int cnt = 0;
        vector<int> vis(26, 25);
        vector<string> tmp(30, "");
        for(auto c: S)
            vis[c-'a'] = cnt++;
        for(auto c: T)
            tmp[vis[c-'a']].push_back(c);
        ostringstream ans;
        for(int i=0;i<=25;i++)
            ans << tmp[i];
        return ans.str();
    }
};